### PR TITLE
Enh/device description

### PR DESCRIPTION
**Pull request by ajtritt**
## Motivation

After #695, Device became a group with a help string. This PR adds a required description field. A Device ObjectMapper was also included for setting a default value for those files generated prior to this change.

This change was inspired by discussions with @ageorgou

## Checklist

- [X] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured the PR description clearly describes problem and the solution?
- [X] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [X] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [X] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

**Review by oruebel — DISMISSED** [review]

**Comment by oruebel** [comment]
Looks like some of tests need to be updated, but other than that it looks good to me.

**Comment by bendichter** [comment]
I prefer these types of additions to be optional. This change will require changes to dozen different conversion scripts and does not offer me much utility. I don't think these types of descriptive fields need to be required

**Comment by ajtritt** [comment]
Then we should just remove Device. A group with a name and a constant help string doesn’t do much.

**Review by bendichter — DISMISSED** [review]
making Device optional obviates the Device io file

**Comment by bendichter** [comment]
The addition of `description` to the tests is now superfluous, but I don't mind leaving them

File: src/pynwb/data/nwb.file.yaml [hunk 1]
```diff
@@ -235,6 +235,10 @@ groups:
           dtype: text
           doc: Value is 'A recording device e.g. amplifier'.
           value: A recording device e.g. amplifier
+        datasets:
```
Comment (bendichter) [inline src/pynwb/data/nwb.file.yaml:240]: Why is this a dataset and not an attribute? I would not expect this to be changed once it is written, so shouldn't it be an attribute (https://github.com/NeurodataWithoutBorders/nwb-schema/issues/48)? Making this required would be a schema-breaking change, so I would like to make this optional.

**Review by bendichter — CHANGES_REQUESTED** [review]

**Pull request closed without merging**